deploy/sre-prometheus/100-node-unschedulable.PrometheusRule.yaml

```
@@ -11,8 +11,8 @@ spec:
   - name: sre-node-unschedulable
     rules:
     - alert: KubeNodeUnschedulableSRE
-      expr: kube_node_spec_unschedulable > 0
-      for: 80m
+      expr: (kube_node_spec_unschedulable > 0 and on(node) kube_node_role{role="master"}) or (kube_node_spec_unschedulable > 0 unless ignoring(node,container,endpoint,job,namespace,service) sum(mapi_machine_set_status_replicas{name=~".*upgrade.*"}) > 0)
```

A minor tweak here - otherwise there is the potential that if a cluster owner decided to create a machine pool with the word `upgrade` in it, it would be excluded from this alert.

The suggestion below will apply if `upgrade` appears at the very end of the machine set name after the availability zone (which is what MUO does). If a cluster owner created a machine pool named `upgrade`, `upgrade` appears before the availability zone.

```suggestion
      expr: (kube_node_spec_unschedulable > 0 and on(node) kube_node_role{role="master"}) or (kube_node_spec_unschedulable > 0 unless ignoring(node,container,endpoint,job,namespace,service) sum(mapi_machine_set_status_replicas{name=~".*upgrade$"}) > 0)
```

---

Is there a label we apply to the upgrade nodes? It would make sense to use that here if we do.

---

I don't believe we label upgrade nodes. We only label the machineset.